### PR TITLE
1.x

### PR DESCRIPTION
**Pull request by Page-**
* Rename to balena
* Add type checking
* Add linting
* Switch to native promises

### CODE REVIEW

File: src/browser.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require('./gs-loader')
 
 var Promise = require('bluebird')
-var TRACKER_NAME = 'resinAnalytics'
+var TRACKER_NAME = 'balenaAnalytics'
```
Comment (Page-) [inline src/browser.js:4]: @gelbal @roman-mazur is this ok to change?

File: src/browser.js [hunk 2]
```diff
@@ -1,50 +1,56 @@
-require('./gs-loader')
+require('./gs-loader');
 
-var Promise = require('bluebird')
-var TRACKER_NAME = 'resinAnalytics'
+var TRACKER_NAME = 'balenaAnalytics';
```
Comment (Page-) [inline src/browser.js]: @gelbal @roman-mazur is this ok to change?

Comment (roman-mazur) [reply src/browser.js]: It might affect how data is displayed on the GS dashboard.
Unfortunately, I think the only way is to try it and see what happens.

Comment (gelbal) [reply src/browser.js]: I found this doc about it: https://www.gosquared.com/docs/javascript-tracking-code/multiple-tracking-codes/

> You can also assign names to different tokens, and then scope actions to a specific project:

Sounds like it's used only as human friendly name in the tracking page. I feel like there is no other purpose but to be sure shall we check with their support?

**Comment by Page-** [comment]
@balena-ci I self-certify!

**Review by ghost — APPROVED** [review]

**Pull request merged**